python/ccxt/base/exchange.py

```diff
@@ -490,9 +490,11 @@ def handle_rest_response(self, response, json_response, url, method='GET', heade
                 self.raise_error(ExchangeNotAvailable, method, url, None, message)
             self.raise_error(ExchangeError, method, url, ValueError('failed to decode json'), response)
 
-    def parse_json(self, http_response):
+    @staticmethod
```

Comment:
There's a couple of issues with this line:

- This change can actually break the `async_support`, which is currently not tested properly (our tests still lack this), can you check for that? 
- Also, it will break transpilation of overrides in derived implementations, if any. Some exchanges, like cryptopia, occasionally spit out a response with an illegal symbol prepended to the json, that might require a pre-parsing stage... 

How about keeping it as an instance method?

Reply:
No async tests!?!

The way I'd go about this is transpiling the synchronous tests to async python, which I can probably do (will require changing the structure of the synchronous tests)

Reply:
> No async tests!?!

Yep (( Still no async-specific tests.

> The way I'd go about this is transpiling the synchronous tests to async python, which I can probably do (will require changing the structure of the synchronous tests)

Yep, that a very good idea, those tests should be transpileable as well, however, I suggest that we do it the other way around, it's easier to transpile from js → py async → py sync, like we do here:

- https://github.com/ccxt/ccxt/blob/master/python/test/test_async.py
- https://github.com/ccxt/ccxt/blob/master/python/test/test.py
- https://github.com/ccxt/ccxt/blob/master/transpile.js#L775

Reply:
So, as you see, we have *some* async tests, but we should probably reorganize the testing framework,  there's very little point in testing twice the parsing part that is common to both the `sync` version and the `async` version. We might want to detach the parsing logic from connectivity logic in those tests.

Reply:
I'm not sure we should implement a full transpillation chain of the tests, because

* A) we may want to test against types or something specific to one language
* B) changing all these files: https://github.com/ccxt/ccxt/tree/master/js/test/Exchange will take a very long time
* C)  `there's very little point in testing twice the parsing part` - the parsing part takes significantly less time than the connectivity part anyway so we go will have to go through a lot of effort for a very small gain

I'm pretty happy with the tests as they are:

* Go throught all the returned fields in depth and check their type in javascript
* Just test connectivity of functions and that they don't throw errors in python and php
* Also we have got async tests already!